docs/custom-flows/error-handling.mdx

```diff
@@ -13,31 +13,67 @@ A user is attempting to sign up with your application, but they are attempting t
 
 <CodeBlockTabs type="framework" options={["React", "JavaScript"]}>
 ```jsx
-import { useState } from "react";
+import { useState, SyntheticEvent } from "react";
 import { useSignUp } from "@clerk/clerk-react";
 
 function Example() {
-  const [error, setError] = useState(null);
-
-  const { signUp } = useSignUp();
-
-  const signUpUser = async () => {
-    const response = await signUp
-      ?.create({
-        identifier: "example@email.com",
-        password: "Tester01!",
-      })
-      .then((res) => console.log("response", res))
-      .catch((err) => setError(err.errors[0].message));
+  const [emailAddress, setEmailAddress] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState<any>();
```

Comment:
```suggestion
  const [error, setError] = useState<Error>();
```

Reply:
I've implemented this change, and it works. But note that TypeScript will complain when accessing properties that are not part of the standard built-in Error object. The error returned by Clerk has extra properties like 'clerkError', 'clerkTraceId', and 'errors'.

Reply:
I wont lie, I forgot that existed

Reply:
Implemented error type checking.

Used `isClerkAPIResponseError` to do type checking on the actual response. This it to avoid changing type of err to any. Details as to why this is preferable here: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#unknown-on-catch-clause-bindings